nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -205,25 +205,50 @@ def __init__(
         else:
             self.out_proj = None
             self._feat_out = d_model
+        self.set_pos_emb_max_len(pos_emb_max_len)
+
+    def set_pos_emb_max_len(self, pos_emb_max_len):
+        """ Sets maximum input length.
+            Pre-calculates internal seq_range mask.
+        """
+        self.pos_emb_max_len = pos_emb_max_len
+        device = next(self.parameters()).device
+        seq_range = torch.arange(0, self.pos_emb_max_len, device=device)
+        if hasattr(self, 'seq_range'):
+            self.seq_range = seq_range
+        else:
+            self.register_buffer('seq_range', seq_range, persistent=False)
+        self.pos_enc.extend_pe(pos_emb_max_len)
 
     @typecheck()
     def forward(self, audio_signal, length=None):
+        max_audio_length: int = audio_signal.size(-1)
```

Comment:
Before this is merged, shouldnt this be audio_signal.max() instead of audio_signal.size(-1) - this is a tensor of shape [B] so size(-1) just gives 32 always? 
Found this while testing the equivalent code for Citrinets. @borisfom @VahidooX

Reply:
1. This is what we used before my changes. 2. This is just to be used in mask creation etc - I don't think it changes any logic there, maybe just makes them bigger than they have to be.  3. Changing it to length.max() would introduce data-dependent shapes (as opposed to shape-dependant shape) and that would almost certainly break TRT.

Reply:
This is in forward() not forward_for_export() so it won't break shapes with TRT. If it was used before I support no need to change it.